platform/mbed_retarget.h

```diff
@@ -66,29 +73,63 @@ namespace mbed {
 
 class FileHandle;
 class DirHandle;
-std::FILE *mbed_fdopen(FileHandle *fh, const char *mode);
+
+/** Bind an mbed FileHandle to a POSIX file descriptor */
+int bind_to_fd(mbed::FileHandle *fh);
+
+/** Unbind an mbed FileHandle from a POSIX file descriptor
+ * Intended for internal use - normally unbinding happens
+ * as part of close() or fclose().
+ */
+int unbind_from_fd(int fd, mbed::FileHandle *fh);
+
+/** Targets may implement this to change stdin, stdout, stderr.
+ *
+ * If the application hasn't provided mbed_override_console, this is called
+ * to give the target a chance to specify a FileHandle for the console.
+ *
+ * If this is not provided or returns NULL, the console will be:
+ *   - UARTSerial if configuration option "platform.stdio-buffered-serial" is
+ *     true and the target has DEVICE_SERIAL;
+ *   - Raw HAL serial via serial_getc and serial_putc if
+ *     "platform.stdio-buffered-serial" is false and the target has DEVICE_SERIAL;
+ *   - stdout/stderr will be a sink and stdin will input a stream of 0s if the
+ *     target does not have DEVICE_SERIAL.
+ *
+ * @param fd file descriptor - STDIN_FILENO, STDOUT_FILENO or STDERR_FILENO
+ * @return  pointer to FileHandle to override normal stream otherwise NULL
+ */
+FileHandle *mbed_target_override_console(int fd);
+
+/** Applications may implement this to change stdin, stdout, stderr.
+ *
+ * This hook gives the application a chance to specify a custom FileHandle
+ * for the console.
+ *
+ * If this is not provided or returns NULL, the console will be specified
+ * by mbed_target_override_console, else will default to serial - see
+ * mbed_target_override_console for more details.
+ *
+ * Example:
+ * @code
+ * FileHandle *mbed_override_console(int) {
+ *     static UARTSerial my_serial(D0, D1);
+ *     return &my_serial;
+ * }
+ * @endcode
+
+ * @param fd file descriptor - STDIN_FILENO, STDOUT_FILENO or STDERR_FILENO
+ * @return  pointer to FileHandle to override normal stream otherwise NULL
+ */
+FileHandle *mbed_override_console(int fd);
```

Comment:
`FileHandle* mbed_override_console` - I added this to the handbook just recently

Reply:
How recently? This is still up:

https://os.mbed.com/docs/latest/reference/guidelines.html#style

* For pointers, `*` is adjacent to a name (`analogin_t *obj`).

Reply:
I should have added the reference, not yet merged: 
https://github.com/ARMmbed/Handbook/pull/382/commits/e2d3f2c94062ac10217395f2b5a6d5b9e1d4a88d (this was not covered there)